### PR TITLE
Eng 7437

### PR DESCRIPTION
**Pull request by akhanzode**

### CODE REVIEW

File: src/frontend/org/voltdb/HTTPClientInterface.java [hunk 1]
```diff
@@ -110,7 +119,34 @@ public void process(Request request, HttpServletResponse response) {
...
+                response.getWriter().print(result);
+            } catch (IllegalStateException | IOException e){
+               // Thrown when we shut down the server via the JSON/HTTP (web studio) API
+               // Essentially we're closing everything down from underneath the HTTP request.
+                m_log.warn("JSON failed to send response: ", e);
+            }
+            return;
+        }
+        //Check if this is resumed request.
+        Object o = request.getAttribute("SQLSUBMITTED");
```
Comment (vtkstef) [inline src/frontend/org/voltdb/HTTPClientInterface.java]: You can shortcut it like 
```java
Boolean.TRUE.equals(request.getAttribute("SQLSUBMITTED"))
```

File: src/frontend/org/voltdb/client/AuthenticatedConnectionCache.java [hunk 2]
```diff
@@ -237,6 +237,7 @@ public synchronized void closeAll()
         if (m_unauthClient != null)
         {
             try {
+                m_unauthClient.drain();
                 m_unauthClient.close();
             } catch (InterruptedException ex) {
                 throw new RuntimeException("Unable to close unauthenticated client.", ex);
```
Comment (vtkstef) [inline src/frontend/org/voltdb/client/AuthenticatedConnectionCache.java]: Catch NoConnectionException too?

File: src/frontend/org/voltdb/client/AuthenticatedConnectionCache.java [hunk 3]
```diff
@@ -245,6 +246,7 @@ public synchronized void closeAll()
         for (Entry<String, Connection> e : m_connections.entrySet())
         {
             try {
+                e.getValue().client.drain();
                 e.getValue().client.close();
             } catch (InterruptedException ex) {
```
Comment (vtkstef) [inline src/frontend/org/voltdb/client/AuthenticatedConnectionCache.java]: Same here?

File: src/frontend/org/voltdb/client/AuthenticatedConnectionCache.java (hunk 2 above)
Comment (akhanzode) [inline src/frontend/org/voltdb/client/AuthenticatedConnectionCache.java]: Its not thrown.

**Pull request closed without merging**